src/main/java/nl/cherement/jak/controller/UserController.java

```diff
@@ -0,0 +1,62 @@
+package nl.cherement.jak.controller;
+
+import nl.cherement.jak.entity.UserEntity;
+import nl.cherement.jak.service.UserPrincipalDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Optional;
+
+@CrossOrigin(origins = "*", allowedHeaders = "*")
+@RestController
+@RequestMapping("/user")
+public class UserController {
+
+
+    @Autowired
+    private UserPrincipalDetailsService userPrincipalDetailsService;
+
+
+    @GetMapping
+    public List<UserEntity> findAll(HttpServletRequest request) {
+        return userPrincipalDetailsService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Optional<UserEntity> find(HttpServletRequest request, @PathVariable("id") Long id) {
+        return userPrincipalDetailsService.findById(id);
+    }
+
+    @GetMapping("/me")
+    public UserEntity auth(Authentication authentication) {
+        return userPrincipalDetailsService.auth(authentication);
+    }
+
+    @PostMapping
+    public UserEntity save(@RequestBody UserDTO userDTO) {
+        return userPrincipalDetailsService.save(userDTO.toEntity());
+    }
+}
+
+class UserDTO extends UserEntity {
+
+    @Override
+    public void setPassword(String password) {
```

Comment:
Het wachtwoord van de User komt hierdoor in plaintext in de database

Reply:
Nee want hij doet in de toEntitity() methode setPassword op de UserEntitiy. Ik overwrite hem hier om te voorkomen dat we hem dubbel encrypten. Namelijk eerst op de DTO en dan op de Entity